hikari/events/message_events.py

```diff
@@ -606,9 +617,40 @@ class MessageDeleteEvent(MessageEvent, abc.ABC):
 
     __slots__: typing.Sequence[str] = ()
 
+    @property
+    def old_message(self) -> typing.Optional[messages.PartialMessage]:
+        """Object of the first deleted message.
+
+        This is contextually useful if you know this is not a bulk deletion
+        event. For all other purposes, this is the same as running
+        `self.old_messages.get(next(iter(self.message_ids)))`.
+
+        Returns
+        -------
+        typing.Optional[hikari.snowflakes.Snowflake]
+            The first deleted message, if in cache, else `builtins.None`.
+        """
+        try:
+            return self.old_messages.get(next(iter(self.message_ids)))
```

Comment:
can't you just do `next(iter(self.old_messages.values()))` here?

Reply:
There is no guarantee that the first value will be the first message, as it might be missing from the cache due to Discord not sending the message create or some other reason

Reply:
In what situation when handling a bulk delete will someone only care about the first message which was deleted